cmake/recipes/external/eigen.cmake

```diff
@@ -61,6 +61,14 @@ if(EIGEN_WITH_MKL)
     )
 endif()
 
+if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" AND ${CMAKE_CXX_COMPILER_VERSION} VERSION_GREATER_EQUAL 12)
+    # Bypass the following GCC bug.
+    # https://gcc.gnu.org/bugzilla/show_bug.cgi?id=106247
+    target_compile_options(Eigen3_Eigen INTERFACE
+        "-Wno-array-bounds"
```

Comment:
Same here I'm tempted to use the warnoff/warnon mechanism rather than having this flag spill into our own codebase. But given how widely we use Eigen that sounds really annoying...

Reply:
I am a bit worried about this due to exactly this reason, but it is worth a try.

Reply:
One option would be to use a "proxy" header to fence the include with warnoff/warnon. We can have one proxy header for Dense/Sparse/Geometry.

Do you know if the warning is triggered when the template is instantiated? (in our own codebase), or when the header is parsed (i.e. inside the Eigen header)?

Reply:
This is triggered in our own code.

Reply:
Ok it seems that this warning is unreliable with gcc. Let's add it as a PRIVATE warning in the main Lagrange targets then (e.g. when we define our target via `lagrange_add_module()`), to avoid spilling out into client code.

Reply:
Can you take a look at 84129f5? It seems there are only a few places that triggers this warning, and we can put warning ignore guards around those places.

Reply:
Nice! That's even better.